v-next/core/src/internal/global-parameters.ts

```diff
@@ -48,21 +58,47 @@ export function buildGlobalParameterMap(
   return globalParametersIndex;
 }
 
-export function buildGlobalParameterDefinition(options: {
+export function buildGlobalParameterDefinition<T extends ParameterType>({
+  name,
+  description,
+  parameterType,
+  defaultValue,
+}: {
   name: string;
   description: string;
-  parameterType: ParameterType;
-  defaultValue: any;
+  parameterType?: T;
```

Comment:
@alcuadrado I think `parameterType` should be just `type` to be consistent with task parameters. wdyt? I'll change it on a follow up PR if you agree.

Reply:
Added an [issue](https://github.com/NomicFoundation/hardhat/issues/5415) to track it.